### PR TITLE
Major revisions to `plot_diags.py`

### PR DESCRIPTION
**Pull request by kevindougherty-noaa**
These revisions involve utilizing `emcpy` to make it much more concise/cleaner to produce plots.

### CODE REVIEW

**Comment by XiujuanSu-NOAA** [comment]
A quick question, are  green lines what you have changed?

**Comment by kevindougherty-noaa** [comment]
@XiujuanSu-NOAA Yes. The red is what was removed, the green is what was changed.

**Review by EdwardSafford-NOAA — COMMENTED** [review]
You weren't kidding about major revisions!  Just curious -- when do you sleep? :) 

Looks good to me and I look forward to working with the overhauled package.

File: pyGSI/plot_diags.py [hunk 1]
```diff
@@ -164,492 +83,417 @@ def _get_obs_type(obs_id):
...
 
-    if len(data) == 0:
-        _no_data_spatial(metadata, outdir, region=region)
-        return
-    else:
-        stats = _calculate_stats(data)
+def _no_data_map(plotmap, domain, metadata):
+    """
+    Creates a plot if there is no data that displays
+    'No Data' across the map.
```
[the remark this answers is not in the document]
Comment (kevindougherty-noaa) [reply pyGSI/plot_diags.py:363]: Sounds a lot better, thanks!

**Review by XiujuanSu-NOAA — COMMENTED** [review]
confused with these lines, the position of "
        180: ("Surface Marine w/ Station Pressure (Ship, Buoy, C-MAN, "
              "Tide Guage)"),
Everything else looks good to me.

**Comment by kevindougherty-noaa** [comment]
@XiujuanSu-NOAA The purpose of some of the strings being separated on different lines is due to PEP8 compliance. PEP8 prefers that each line be less than 79 characters.

**Comment by XiujuanSu-NOAA** [comment]
Thanks

**Comment by kevindougherty-noaa** [comment]
@XiujuanSu-NOAA If you could approve the changes if all good, then I can merge these changes and then close the PR.

**Comment by XiujuanSu-NOAA** [comment]
Yes, I approve.  I try to find on the gethub to approve your changes.

Su

**Review by CoryMartin-NOAA — APPROVED** [review]
Removed a couple hundred lines of code, nice work @kevindougherty-noaa

**Comment by kevindougherty-noaa** [comment]
@CoryMartin-NOAA and probably added a couple hundred more doc lines lol. Thanks for approving.

**Pull request merged**